modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md

```diff
@@ -323,58 +577,95 @@ If the value is not null than it will be transformed to the specified `Either.Le
 
 Example:
 
-```kotlin:ank
-"value".rightIfNull { "left" }
+```kotlin:ank:playground
+import arrow.core.rightIfNull
+
+val value =
+//sampleStart
+ "value".rightIfNull { "left" }
+//sampleEnd
+fun main() {
+ println(value)
+}
 ```
 
-```kotlin:ank
-null.rightIfNull { "left" }
+```kotlin:ank:playground
+import arrow.core.rightIfNull
+
+val value =
+//sampleStart
+ null.rightIfNull { "left" }
+//sampleEnd
+fun main() {
+ println(value)
+}
 ```
 
- Arrow contains `Either` instances for many useful typeclasses that allows you to use and transform right values.
- Both Option and Try don't require a type parameter with the following functions, but it is specifically used for Either.Left
+Arrow contains `Either` instances for many useful typeclasses that allows you to use and transform right values.
+Both Option and Try don't require a type parameter with the following functions, but it is specifically used for Either.Left
 
- [`Functor`]({{ '/docs/arrow/typeclasses/functor/' | relative_url }})
+[`Functor`]({{ '/docs/arrow/typeclasses/functor/' | relative_url }})
 
- Transforming the inner contents
+Transforming the inner contents
 
-```kotlin:ank
-import arrow.core.extensions.either.functor.*
+```kotlin:ank:playground
+import arrow.core.Right
 
-Right(1).map {it + 1}
+val value =
+//sampleStart
+ Right(1).map{ it + 1 }
+//sampleEnd
+fun main() {
+ println(value)
+}
 ```
 
- [`Applicative`]({{ '/docs/arrow/typeclasses/applicative/' | relative_url }})
+[`Applicative`]({{ '/docs/arrow/typeclasses/applicative/' | relative_url }})
 
- Computing over independent values
+Computing over independent values
 
-```kotlin:ank
-import arrow.core.extensions.either.apply.*
+```kotlin:ank:playground
+import arrow.core.Either
+import arrow.core.extensions.either.apply.tupled
 
-tupled(Either.Right(1), Either.Right("a"), Either.Right(2.0))
+val value =
+//sampleStart
+ tupled(Either.Right(1), Either.Right("a"), Either.Right(2.0))
+//sampleEnd
+fun main() {
+ println(value)
+}
 ```
 
- [`Monad`]({{ '/docs/arrow/typeclasses/monad/' | relative_url }})
+[`Monad`]({{ '/docs/arrow/typeclasses/monad/' | relative_url }})
+
+Computing over dependent values ignoring absence
 
- Computing over dependent values ignoring absence
 
-```kotlin
+```kotlin:ank:playground
 import arrow.core.extensions.fx
+import arrow.core.Either
 
-Either.fx {
+val value =
+//sampleStart
+ Either.fx<Int, Int> {
   val (a) = Either.Right(1)
   val (b) = Either.Right(1 + a)
   val (c) = Either.Right(1 + b)
   a + b + c
+ }
+//sampleEnd
+fun main() {
+ println(value)
 }
-// Right(6)
 ```
 
 ### Supported type classes
 
 ```kotlin:ank:replace
-import arrow.reflect.*
-import arrow.core.*
+import arrow.reflect.DataType
+import arrow.reflect.tcMarkdownList
+import arrow.core.Either
 
 DataType(Either::class).tcMarkdownList()
 ```
```

Comment:
This guy was missing completely from our docs :) 
```suggestion
```

ank_macro_hierarchy(arrow.core.Either)
```